#  See http://gcc.gnu.org/PR11587
#  Fix SH ICE in reload_cse_simplify_operands
#  Michael Eager (eager@mvista.com)

diff -urNp gcc-3.3.1-orig/gcc/config/sh/sh.c gcc-3.3.1/gcc/config/sh/sh.c
--- gcc-3.3.1-orig/gcc/config/sh/sh.c	Fri Aug 29 15:19:26 2003
+++ gcc-3.3.1/gcc/config/sh/sh.c	Fri Aug 29 15:22:49 2003
@@ -7870,7 +7870,8 @@ sh_register_move_cost (mode, srcclass, d
 	   * ((GET_MODE_SIZE (mode) + 7) / 8U));
 
   if ((dstclass == FPUL_REGS
-       && REGCLASS_HAS_GENERAL_REG (srcclass))
+       && (REGCLASS_HAS_GENERAL_REG (srcclass))
+	|| srcclass ==T_REGS)
       || (srcclass == FPUL_REGS
 	  && REGCLASS_HAS_GENERAL_REG (dstclass)))
     return 5;
